core/trino-main/src/main/java/io/trino/execution/resourcegroups/InternalResourceGroupManager.java

```diff
@@ -273,7 +273,7 @@ public int getQueriesQueuedOnInternal()
     private static int getQueriesQueuedOnInternal(InternalResourceGroup resourceGroup)
     {
         if (resourceGroup.subGroups().isEmpty()) {
-            return Math.min(resourceGroup.getQueuedQueries(), resourceGroup.getSoftConcurrencyLimit() - resourceGroup.getRunningQueries());
+            return Math.max(Math.min(resourceGroup.getQueuedQueries(), resourceGroup.getSoftConcurrencyLimit() - resourceGroup.getRunningQueries()), 0);
```

Comment:
We can use the Guava function `Ints.constrainToRange()` for this.

Reply:
Let me address that and update my PR.

Reply:
I think `Ints.constrainRange` logic is a bit different it does Math.min(Math.max so I still have to use Math.max